src/Scenario/PresenceMonitoringScenario/PresenceMonitoringScenario.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TelegramOSINT\Scenario\PresenceMonitoringScenario;
+
+use TelegramOSINT\Client\AuthKey\AuthKeyCreator;
+use TelegramOSINT\Client\StatusWatcherClient\Models\HiddenStatus;
+use TelegramOSINT\Client\StatusWatcherClient\Models\ImportResult;
+use TelegramOSINT\Client\StatusWatcherClient\Models\User;
+use TelegramOSINT\Client\StatusWatcherClient\StatusWatcherCallbacks;
+use TelegramOSINT\Client\StatusWatcherClient\StatusWatcherClient;
+use TelegramOSINT\LibConfig;
+use TelegramOSINT\Scenario\ClientGenerator;
+use TelegramOSINT\Scenario\ScenarioInterface;
+
+class PresenceMonitoringScenario implements ScenarioInterface, StatusWatcherCallbacks
+{
+    /**
+     * @var StatusWatcherClient
+     * */
+    private $client;
+    /**
+     * @var string
+     * */
+    private $authKey;
+    /**
+     * @var string[]
+     * */
+    private $numbers;
+    /**
+     * @var PresenceMonitoringCallbacks
+     * */
+    private $callbacks;
+
+
+    /**
+     * PresenceMonitoringScenario constructor.
+     *
+     * @param array $numbers
+     * @param PresenceMonitoringCallbacks $callbacks
+     * @throws \TelegramOSINT\Exception\TGException
+     */
+    public function __construct(
+        array $numbers,
+        PresenceMonitoringCallbacks $callbacks
+    ) {
+        $generator = new ClientGenerator(LibConfig::ENV_AUTHKEY);
```

Comment:
Let's pass generator as ctor arg, it will make the scenario testable.

Reply:
right

Reply:
need help here